ci-operator/step-registry/cucushift/installer/check/aws/rootvolume/cucushift-installer-check-aws-rootvolume-commands.sh

```diff
@@ -0,0 +1,231 @@
+
+# Determine expected edge size
+EXPECTED_EDGE_SIZE=""
+if [[ -n "${EDGE_SIZE}" ]]; then
+  EXPECTED_EDGE_SIZE="${EDGE_SIZE}"
+elif [[ -n "${EDGE_THROUGHPUT}" ]]; then
+  # Edge pool exists but size not explicitly set, use default
+  EXPECTED_EDGE_SIZE="${DEFAULT_SIZE}"
+fi
+
+
+echo "-------------------------------------------------------------"
+echo "Expected root volume configuration"
+echo "-------------------------------------------------------------"
+echo "Configuration priority: pool-specific settings > defaultMachinePlatform"
+echo ""
+echo "Worker rootVolume:"
+echo "  size: ${EXPECTED_COMPUTE_SIZE} GiB"
+echo "  throughput: ${EXPECTED_COMPUTE_THROUGHPUT} MiB/s"
+if [[ -n "${COMPUTE_THROUGHPUT}" ]]; then
+  echo "  (from compute[0].platform.aws.rootVolume)"
+elif [[ -n "${DEFAULT_THROUGHPUT}" ]]; then
+  echo "  (from platform.aws.defaultMachinePlatform.rootVolume)"
+fi
+echo "Control-plane rootVolume:"
+echo "  size: ${EXPECTED_CONTROL_PLANE_SIZE} GiB"
+echo "  throughput: ${EXPECTED_CONTROL_PLANE_THROUGHPUT} MiB/s"
+if [[ -n "${CONTROL_PLANE_THROUGHPUT}" ]]; then
+  echo "  (from controlPlane.platform.aws.rootVolume)"
+elif [[ -n "${DEFAULT_THROUGHPUT}" ]]; then
+  echo "  (from platform.aws.defaultMachinePlatform.rootVolume)"
+fi
+echo "Edge rootVolume:"
+echo "  size: ${EXPECTED_EDGE_SIZE} GiB"
+echo "  throughput: ${EXPECTED_EDGE_THROUGHPUT} MiB/s"
+if [[ -n "${EDGE_THROUGHPUT}" ]]; then
+  echo "  (from compute[1].platform.aws.rootVolume)"
+elif [[ -n "${DEFAULT_THROUGHPUT}" ]]; then
+  echo "  (from platform.aws.defaultMachinePlatform.rootVolume)"
+fi
+
+ret=0
+declare -a FAILURE_SUMMARY=()
+
+function log_machine_root_volume_spec() {
+  local node_name="$1"
+  local machine_ref="$2"
+
+  if [ -z "${machine_ref}" ]; then
+    return
+  fi
+
+  local machine_ns="${machine_ref%%/*}"
+  local machine_name="${machine_ref##*/}"
+  local spec_type spec_size spec_throughput
+
+  spec_type=$(oc get machine -n "${machine_ns}" "${machine_name}" -o jsonpath='{.spec.providerSpec.value.rootVolume.type}' 2>/dev/null || true)
+  spec_size=$(oc get machine -n "${machine_ns}" "${machine_name}" -o jsonpath='{.spec.providerSpec.value.rootVolume.size}' 2>/dev/null || true)
+  spec_throughput=$(oc get machine -n "${machine_ns}" "${machine_name}" -o jsonpath='{.spec.providerSpec.value.rootVolume.throughput}' 2>/dev/null || true)
+
+  if [[ -n "${spec_type}" ]] || [[ -n "${spec_size}" ]] || [[ -n "${spec_throughput}" ]]; then
+    echo "INFO: ${node_name} desired rootVolume spec (machine ${machine_ref}) type=${spec_type} size=${spec_size}GiB throughput=${spec_throughput}MiB/s"
+  fi
+}
+
+function verify_nodes() {
+  local role_label="$1"
+  local role_name="$2"
+  local expected_throughput="$3"
+
+  local nodes
+  nodes=$(oc get nodes -l "${role_label}" -o jsonpath='{.items[*].metadata.name}')
+  if [ -z "${nodes}" ]; then
+    echo "WARNING: No ${role_name} nodes found, skipping"
+    return
```

Comment:
So if no _${role_name} nodes_ were found, the verification is PASS, is that right?

Reply:
I added the `required` parameter to indicate whether the "node not found" is an error or not. Control Plane not found should be an error.